examples/labs/dashboard/main.py

```diff
@@ -0,0 +1,23 @@
+from pitop import Pitop, LED, Button, UltrasonicSensor
+from pitop.labs import WebController
+from time import sleep
+import gevent
+
+robot = Pitop()
+robot.add_component(Button('D1'))
+robot.add_component(LED('D2'))
+robot.add_component(UltrasonicSensor('D3'))
+
+dashboard_server = WebController()
+
+
+def broadcast_state():
+    while True:
```

Comment:
It would be good to add a helper function to broadcast something on loop at a specified frequency in a thread - it seems like something that could be very helpful for many applications. I used a scheduler [here](https://github.com/pi-top/pi-top-Python-SDK/pull/334/commits/c5fa9f1e3d5a4ae52335d4ea60d5617fe44c1ea3) for this sort of thing which we could probably hide away in the background.

Out of interest, is there a good reason to use gevents here instead of normal python threading? For this application, I guess it doesn't really matter as it's only one thread with a very simple task, but I would always lean to using standard Python libs where possible.

Reply:
I think adding concurrency helpers is out of scope here, we should talk about what those would look like in a meeting if you want to schedule that.

In terms of why I used gevent it's because the underlying WSGIServer is a gevent server which doesn't always play nice with std-lib threading. The underlying server was added before this PR so I'm not 100% of the tradeoffs and limitations there. For now I'll add to the docs that users should use gevent threads while using a WebServer.

Certainly this is something we should dig into more thoroughly before we move this into the main SDK

Reply:
I've added a warning to the docs about multithreading 👍